src/app/post-page/post-page.component.ts

```diff
@@ -62,34 +115,90 @@ export class PostPageComponent implements OnInit {
     },
     (err) => {
       // fill in this part...
+      this._router.navigate(['/pageNotFound']); // ?
       console.log(err);
     })
   }
+  
+  // Not working
+  // checkAnonymous() {
+  //   const name = this.commentForm.get('name');
+  //   this.isAnonymous = true;
+  //   this.isReveal = false;
+  //   name.clearValidators();
+  //   name.setValue('');
+  //   name.disable();
+  //   name.updateValueAndValidity();
+  // }
+
+  // checkReveal() {
+  //   const name = this.commentForm.get('name');
+  //   this.isReveal = true;
+  //   this.isAnonymous = false;
+  //   name.setValidators(Validators.required);
+  //   name.enable();
+  //   name.updateValueAndValidity();
+  // }
+
+  // checkRevealAlt() {
+  //   const name = this.commentForm.get('name');
+  //   this.isReveal = true;
+  //   this.isAnonymous = false;
+  //   name.setValidators(Validators.required);
+  //   name.enable();
+  //   name.updateValueAndValidity();
+  // }
 
   goPrevious() {
-    let previousId = this.post.id - 1;
-    this._router.navigate(['home/post'], { queryParams: { id: previousId } }); // is the route correct?
+    let previousId = (this.post.id - 1) <= 0 ? this.numberOfApprovedPosts : this.post.id - 1;
+    // this._router.navigate(['home/post'], { queryParams: { id: previousId } });
+    this._router.navigate(['./'], 
+                          { relativeTo: this._route, 
+                            queryParams: { id: previousId }}); // relative navigation
   }
 
   goNext() {
-    let nextId = this.post.id + 1;
-    this._router.navigate(['home/post'], { queryParams: { id: nextId } }); // is the route correct?
+    let nextId = (this.post.id + 1) >= this.numberOfApprovedPosts ? 1 : this.post.id + 1;
+    // this._router.navigate(['home/post'], { queryParams: { id: nextId } });
+    this._router.navigate(['./'], 
+                          { relativeTo: this._route, 
+                            queryParams: { id: nextId }}); // relative navigation
   }
 
   updatePostLikes() {
-    this._postService.updateLikes(this.post.id, this.post.likes + 1).subscribe(); // subscribe need para?
+    this._postService.updateLikes(this.post.id, this.post.likes + 1)
+      .subscribe(() => { this.reloadData; });
```

Comment:
this.reloadData is a method... btw while u have to update the database as per usual, u do not have to reload the whole data for just one single like right? u can simply add one to the number of likes on the frontend (to make it seem like the user has liked the post), and then when the user reloads the data the number of likes will of course be updated.